src/Data/Bin/Properties.agda

```diff
@@ -1,143 +1,904 @@
 ------------------------------------------------------------------------
 -- The Agda standard library
 --
--- Properties of the binary representation of natural numbers
+-- Arithmetic properties related to addition and multiplication of
+-- binary represented natural numbers.
 ------------------------------------------------------------------------
 
 {-# OPTIONS --without-K --safe #-}
 
 module Data.Bin.Properties where
 
-open import Data.Bin
-open import Data.Digit using (Bit; Expansion)
-import Data.Fin as Fin
-import Data.Fin.Properties as 𝔽ₚ
-open import Data.List.Base using (List; []; _∷_)
-open import Data.List.Properties using (∷-injective)
-open import Data.Nat
-  using (ℕ; zero; z≤n; s≤s)
-  renaming (suc to 1+_; _+_ to _+ℕ_; _*_ to _*ℕ_; _≤_ to _≤ℕ_)
-import Data.Nat.Properties as ℕₚ
-open import Data.Product using (proj₁; proj₂)
-open import Function using (_∘_)
-open import Relation.Binary
-open import Relation.Binary.Consequences
-open import Relation.Binary.PropositionalEquality
-  using (_≡_; _≢_; refl; sym; isEquivalence; resp₂; decSetoid)
-open import Relation.Nullary using (yes; no)
+open import Algebra using (Magma; Semigroup; Monoid; CommutativeMonoid;
```

Comment:
I think at this point you can open `Algebra` and `Algebra.Structures` without listing all the individual modules.

Reply:
Done.